skrf/network.py

```diff
@@ -3349,6 +3349,7 @@ def renumber(self, from_ports: Sequence[int], to_ports: Sequence[int], only_z0:
         if self.port_names is not None:
             self.port_names = np.array(self.port_names)
             self.port_names[to_ports] = self.port_names[from_ports]
+        self.port_names = list(self.port_names)
```

Comment:
```suggestion
            _port_names = np.array(self.port_names)
            _port_names[to_ports] = _port_names[from_ports]
            self.port_names = _port_names.tolist()
```

I think this should pass the tests. The issue was, that `self.port_names` was intended to be a list but did change to a `np.ndarray` in this step.